firefox-ios/Client/Frontend/PasswordGenerator/PasswordGeneratorViewController.swift

```diff
@@ -0,0 +1,293 @@
+        imageView.image = UIImage(named: StandardImageIdentifiers.Large.login)?.withRenderingMode(.alwaysTemplate)
+        imageView.accessibilityIdentifier = AccessibilityIdentifiers.PasswordGenerator.headerImage
+    }
+    private lazy var descriptionLabel: UILabel = .build { label in
+        label.numberOfLines = 0
+        label.font = FXFontStyles.Regular.subheadline.scaledFont()
+        label.text = .PasswordGenerator.Description
+        label.accessibilityIdentifier = AccessibilityIdentifiers.PasswordGenerator.descriptionLabel
+    }
+
+    private lazy var header: UIView = .build { view in
+        view.accessibilityIdentifier = AccessibilityIdentifiers.PasswordGenerator.header
+    }
+
+    private lazy var passwordField: UIView = .build { field in
+        field.layer.borderWidth = UX.passwordFieldBorderWidth
+        field.layer.cornerRadius = UX.passwordFieldCornerRadius
+        field.accessibilityIdentifier = AccessibilityIdentifiers.PasswordGenerator.passwordField
+    }
+
+    private lazy var passwordLabel: UILabel = .build { label in
+        label.accessibilityIdentifier = AccessibilityIdentifiers.PasswordGenerator.passwordlabel
+    }
+
+    private lazy var passwordRefreshButton: UIButton = .build { button in
+        button.setImage(
+            UIImage(named: StandardImageIdentifiers.Large.arrowClockwise)?.withRenderingMode(.alwaysTemplate), for: .normal)
+        button.contentHorizontalAlignment = .fill
+        button.contentVerticalAlignment = .fill
+        button.accessibilityLabel = .PasswordGenerator.RefreshPasswordButtonA11yLabel
+        button.accessibilityIdentifier = AccessibilityIdentifiers.PasswordGenerator.passwordRefreshButton
+    }
+
+    private lazy var usePasswordButton: PrimaryRoundedButton = .build()
+
+    // MARK: - Initializers
+
+    init(windowUUID: WindowUUID,
+         themeManager: ThemeManager = AppContainer.shared.resolve(),
+         notificationCenter: NotificationProtocol = NotificationCenter.default,
+         currentTab: Tab) {
+        self.windowUUID = windowUUID
+        self.themeManager = themeManager
+        self.notificationCenter = notificationCenter
+        self.passwordGeneratorState = PasswordGeneratorState(windowUUID: windowUUID)
+        super.init(nibName: nil, bundle: nil)
+        self.subscribeToRedux()
+    }
+
+    deinit {
+        unsubscribeFromRedux()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - View Lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        listenForThemeChange(view)
+        buildPasswordLabel()
+        buildUsePasswordButton()
+        setupView()
+        applyTheme()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        applyTheme()
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.3) {
```

Comment:
curious - why do we need to dispatch here?

Reply:
I kept encountering a problem where the voice reader would prioritize reading the close button of the bottomsheet over the Header of the password generator. This dispatch delays the notification, which fixes that issue